### PR TITLE
Fix conventions for object naming / use gson for expiry date parsing

### PR DESCRIPTION
**Pull request by guperrot**
Please have a look at our [guidelines for contributions](https://github.com/Microsoft/AppCenter-SDK-Android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

* Fix conventions for object naming.
* Just use Gson for date parsing in token.

### CODE REVIEW

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java [hunk 1]
```diff
@@ -81,7 +81,7 @@
 
     private Map<DefaultAppCenterFuture<?>, ServiceCall> mPendingCalls = new HashMap<>();
 
-    private final Map<String, ServiceCall> mOutgoingPendingOperationCalls = new HashMap<>();
+    private final HashMap<String, ServiceCall> mOutgoingPendingOperationCalls = new HashMap<>();
```
Comment (guperrot) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: should not put null value on Map as its implementation dependent so we get a lint warning. By telling we depend on this implementation we can depend on ability to set null values

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/LocalDocumentStorage.java [hunk 2]
```diff
@@ -338,15 +306,47 @@ void updatePendingOperation(PendingOperation operation) {
...
-        return getContentValues(
-                operation.getPartition(),
-                operation.getDocumentId(),
-                operation.getDocument(),
-                operation.getEtag(),
-                operation.getExpirationTime(),
-                now,
-                now,
-                operation.getOperation());
+    @NonNull
```
Comment (guperrot) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/LocalDocumentStorage.java]: Someone didn't run the formatter, the formatter can reorder methods based on static or visiblity

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java (hunk 1 above)
Comment (dhei) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: +1.

Plus `mOutgoingPendingOperationCalls` is only a private property so it's not leaking anything and there's no point to use `Map` here.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/LocalDocumentStorage.java [hunk 3]
```diff
@@ -141,7 +153,7 @@ void resetDatabase() {
                 document.getPartition(),
                 document.getId(),
                 document.toString(),
-                document.getEtag(),
+                document.getETag(),
```
Comment (yochaude) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/LocalDocumentStorage.java]: I updated the specs in the wiki to match the casing.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java (hunk 1 above)
Comment (guperrot) [reply sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java]: That was my feedback to use Map as this is the pattern to use interface whenever possible. This is an exception.

File: sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/Page.java [hunk 4]
```diff
@@ -16,18 +16,18 @@
...
      */
     @SerializedName(value = Constants.DOCUMENTS_FIELD_NAME)
-    private List<Document<T>> documents;
+    private List<Document<T>> mItems;
 
     /**
      * Document error.
      */
-    private DocumentError error;
+    private transient DocumentError mError;
```
Comment (dhei) [inline sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/models/Page.java]: Good catch 👍

**Pull request merged**